import "galaxy.proto";
package baidu.galaxy;

option cc_generic_services = true;

message EnterSafeModeRequest {

}
message EnterSafeModeResponse {

}
message LeaveSafeModeRequest {

}
message LeaveSafeModeResponse {

}
message StatusRequest {

}
message StatusResponse {

}

message KeepAliveRequest {

}
message KeepAliveResponse {

}

message CreateContainerRequest {
}

message CreateContainerResponse {
}

message RemoveContainerRequest {
}

message RemoveContainerResponse {
}

//rpc ResizeContainer(ResizeContainerRequest) returns (ResizeContainerResponse);

message ListContainerRequest {
}

message ListContainersResponse {
}

message ShowContainerRequest {

}

message ShowContainerResponse {
}

message CleaseContainerRequest {
}

message ClearContainerResponse {
}

message AddAgentRequest {

}
message AddAgentResponse {

}
message RemoveAgentRequest {

}
message RemoveAgentResponse {

}
message OnlineAgentRequest {

}
message OnlineAgentResponse {

}
message OfflineAgentRequest {

}
message OfflineAgentResponse {

}
message ListAgentsRequest {

}
message ListAgentsResponse {

}
message CreateTagRequest {

}
message CreateTagResponse {

}
message RemoveTagRequest {

}
message RemoveTagResponse {

}
message ListTagsRequest {

}
message ListTagsResponse {

}
message AddAgentToTagRequest {

}
message AddAgentToTagResponse {

}
message RemoveAgentFromTagRequest {

}
message RemoveAgentFromTagResponse {

}
message ListAgentsByTagRequest {

}
message ListAgentsByTagResponse {

}
message AddUserRequest {

}
message AddUserResponse {

}
message RemoveUserRequest {

}
message RemoveUserResponse {

}
message ListUsersRequest {

}
message ListUsersResponse {

}
message ShowUserRequest {

}
message ShowUserResponse {

}
message GrantUserRequest {

}
message GrantUserResponse {

}
message AssignQuotaRequest {

}
message AssignQuotaResponse {

}

message CreateContainerGroupRequest {
}

message CreateContainerGroupResponse {
}

message RemoveContainerGroupRequest {
}

message RemoveContainerGroupResponse {
}

message UpdateContainerGroupRequest {
} 

message UpdateContainerGroupResponse {}


message ListContainerGroupsRequest {}
message ListContainerGroupsResponse {}
message ShowContainerGroupRequest {} 
message ShowContainerGroupResponse {}
message ClearContainerRequest {}
message GetTagsByAgentRequest {}
message GetTagsByAgentResponse {}
message AddAgentToPoolRequest {}
message AddAgentToPoolResponse {}
message RemoveAgentFromPoolRequest {}
message RemoveAgentFromPoolResponse {}
message ListAgentsByPoolRequest {}
message ListAgentsByPoolResponse {}
message GetPoolByAgentRequest {}
message GetPoolByAgentResponse {}

service ResMan {

    //op api
    rpc EnterSafeMode(EnterSafeModeRequest) returns (EnterSafeModeResponse);
    rpc LeaveSafeMode(LeaveSafeModeRequest) returns (LeaveSafeModeResponse);

    rpc Status(StatusRequest) returns (StatusResponse);

    //internal RPC 
    rpc KeepAlive(KeepAliveRequest) returns (KeepAliveResponse);

    //container
    rpc CreateContainerGroup(CreateContainerGroupRequest) returns (CreateContainerGroupResponse);
    rpc RemoveContainerGroup(RemoveContainerGroupRequest) returns (RemoveContainerGroupResponse);

    // modify replica  only?
    rpc UpdateContainerGroup(UpdateContainerGroupRequest) returns (UpdateContainerGroupResponse);

    rpc ListContainerGroups(ListContainerGroupsRequest) returns (ListContainerGroupsResponse);
    rpc ShowContainerGroup(ShowContainerGroupRequest) returns (ShowContainerGroupResponse);

    // stop all the process in container, [volum & data]
    // [depress]
    //rpc StopContainerGroup(StopContainerGroupRequest) returns (StopContainerGroupResponse);

    // op 
    rpc CreateContainer(CreateContainerRequest) returns (CreateContainerResponse);
    rpc RemoveContainer(RemoveContainerRequest) returns (RemoveContainerResponse);
    rpc ShowContainer(ShowContainerRequest) returns (ShowContainerResponse);
    rpc ClearContainer(ClearContainerRequest) returns (ClearContainerResponse);
  
    //agent man
    rpc AddAgent(AddAgentRequest) returns (AddAgentResponse);
    rpc RemoveAgent(RemoveAgentRequest) returns (RemoveAgentResponse);
    rpc OnlineAgent(OnlineAgentRequest) returns (OnlineAgentResponse);
    rpc OfflineAgent(OfflineAgentRequest) returns (OfflineAgentResponse);
    rpc ListAgents(ListAgentsRequest) returns (ListAgentsResponse);

    //tag man
    rpc AddAgentsToTag(CreateTagRequest) returns (CreateTagResponse);
    rpc ListTags(ListTagsRequest) returns (ListTagsResponse);
   
    // tag <--> agent
    rpc ListAgentsByTag(ListAgentsByTagRequest) returns (ListAgentsByTagResponse);
    rpc GetTagsByAgent(GetTagsByAgentRequest) returns (GetTagsByAgentResponse);

    rpc AddAgentToPool(AddAgentToPoolRequest) returns (AddAgentToPoolResponse);
    rpc RemoveAgentFromPool(RemoveAgentFromPoolRequest) returns (RemoveAgentFromPoolResponse);
    rpc ListAgentsByPool(ListAgentsByPoolRequest) returns (ListAgentsByPoolResponse);
    rpc GetPoolByAgent(GetPoolByAgentRequest) returns (GetPoolByAgentResponse);

    // user man
    rpc AddUser(AddUserRequest) returns (AddUserResponse);
    rpc RemoveUser(RemoveUserRequest) returns (RemoveUserResponse);
    rpc ListUsers(ListUsersRequest) returns (ListUsersResponse);
    rpc ShowUser(ShowUserRequest) returns (ShowUserResponse);
    rpc GrantUser(GrantUserRequest) returns (GrantUserResponse);
    rpc AssignQuota(AssignQuotaRequest) returns (AssignQuotaResponse);

}



